error.go

```diff
@@ -63,149 +71,180 @@ var (
 )
 
 type (
-	// Error contains the details of a error response.
-	Error struct {
+	// ErrorClass is an error generating function.
+	// It accepts a message and optional key value pairs and produces errors that implement
+	// ServiceError.
+	// If the message is a string or a Stringer then the string value is used.
+	// If the message is an error then the string returned by Error() is used.
+	// Otherwise the string produced using fmt.Sprintf("%v") is used.
+	// The optional key value pairs are intended to provide additional contextual information
+	// and are returned to the client.
+	ErrorClass func(message interface{}, keyvals ...interface{}) error
+
+	// ServiceError is the interface implemented by all errors created using a ErrorClass
+	// function.
+	ServiceError interface {
+		// ResponseStatus dictates the status used to build the response sent to the client.
+		ResponseStatus() int
+		// Token is a unique value associated with the occurrence of the error.
+		Token() string
+	}
+
+	// errorResponse contains the details of a error response. It implements ServiceError.
+	errorResponse struct {
+		// ID is the unique error instance identifier.
+		ID string `json:"id" xml:"id" form:"id"`
 		// Code identifies the class of errors.
 		Code string `json:"code" xml:"code" form:"code"`
 		// Status is the HTTP status code used by responses that cary the error.
 		Status int `json:"status" xml:"status" form:"status"`
 		// Detail describes the specific error occurrence.
 		Detail string `json:"detail" xml:"detail" form:"detail"`
-		// MetaValues contains additional key/value pairs useful to clients.
-		MetaValues map[string]interface{} `json:"meta,omitempty" xml:"meta,omitempty" form:"meta,omitempty"`
+		// Meta contains additional key/value pairs useful to clients.
+		Meta []map[string]interface{} `json:"meta,omitempty" xml:"meta,omitempty" form:"meta,omitempty"`
```

Comment:
this design choice surprises me, why make it a slice of kvs instead of a single map?